huaweicloud/resource_huaweicloud_networking_secgroup.go

```diff
@@ -102,108 +100,107 @@ func ResourceNetworkingSecGroupV2() *schema.Resource {
 			Description: &description,
 		}
-		_, err = groups.Update(networkingClient, d.Id(), updateOpts).Extract()
+		_, err = groups.Update(client, d.Id(), updateOpts)
 		if err != nil {
-			return fmtp.Errorf("Error updating description of security group %s: %s", d.Id(), err)
+			return fmtp.DiagErrorf("Error updating description of security group %s: %s", d.Id(), err)
 		}
 	}
 
 	// Delete the default security group rules if it has been requested.
 	deleteDefaultRules := d.Get("delete_default_rules").(bool)
 	if deleteDefaultRules {
 		for _, rule := range securityGroup.SecurityGroupRules {
-			if err := rules.Delete(networkingClient, rule.ID).ExtractErr(); err != nil {
-				return fmtp.Errorf(
+			if err := rules.Delete(client, rule.ID).ExtractErr(); err != nil {
+				return fmtp.DiagErrorf(
 					"There was a problem deleting a default security group rule: %s", err)
 			}
 		}
 	}
 
-	return resourceNetworkingSecGroupV2Read(d, meta)
+	return resourceNetworkingSecGroupV3Read(ctx, d, meta)
 }
 
-func resourceNetworkingSecGroupV2Read(d *schema.ResourceData, meta interface{}) error {
+func resourceNetworkingSecGroupV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
-	segClient, err := config.SecurityGroupV1Client(GetRegion(d, config))
+	client, err := config.NetworkingV3Client(GetRegion(d, config))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud networking client: %s", err)
+		return fmtp.DiagErrorf("Error creating HuaweiCloud networking v3 client: %s", err)
 	}
 
 	logp.Printf("[DEBUG] Retrieve information about security group: %s", d.Id())
-	secGroup, err := securitygroups.Get(segClient, d.Id()).Extract()
+	securityGroup, err := groups.Get(client, d.Id())
 	if err != nil {
-		return CheckDeleted(d, err, "HuaweiCloud Security group")
+		return common.CheckDeletedDiag(d, err, "HuaweiCloud Security group")
 	}
 
-	logp.Printf("[DEBUG] Retrieved Security Group %s: %+v", d.Id(), secGroup)
+	logp.Printf("[DEBUG] Retrieved Security Group %s: %+v", d.Id(), securityGroup)
 
 	mErr := multierror.Append(nil,
 		d.Set("region", GetRegion(d, config)),
-		d.Set("name", secGroup.Name),
-		d.Set("description", secGroup.Description),
-		d.Set("enterprise_project_id", secGroup.EnterpriseProjectId),
-		d.Set("rules", flattenSecurityGroupRules(secGroup)),
+		d.Set("name", securityGroup.Name),
+		d.Set("description", securityGroup.Description),
+		d.Set("enterprise_project_id", securityGroup.EnterpriseProjectId),
+		d.Set("rules", flattenSecurityGroupRules(securityGroup)),
+		d.Set("created_at", securityGroup.CreatedAt),
+		d.Set("updated_at", securityGroup.UpdatedAt),
 	)
 	if mErr.ErrorOrNil() != nil {
-		return mErr
+		return diag.FromErr(mErr)
 	}
 
 	return nil
 }
 
-func flattenSecurityGroupRules(secGroup *securitygroups.SecurityGroup) []map[string]interface{} {
+func flattenSecurityGroupRules(secGroup *groups.SecurityGroup) []map[string]interface{} {
```

Comment:
please merge it with `flattenSecurityGroupDataRules` function

Reply:
merged